src/or/proto_socks.c

```diff
@@ -88,13 +88,16 @@ socks_request_free_(socks_request_t *req)
 
 static int
 parse_socks4_request(const uint8_t *raw_data, socks_request_t *req,
-                     size_t datalen, int *is_socks4a)
+                     size_t datalen, int *is_socks4a, size_t *drain_out)
 {
   // http://ss5.sourceforge.net/socks4.protocol.txt
   // http://ss5.sourceforge.net/socks4A.protocol.txt
   int res = 1;
   tor_addr_t destaddr;
 
+  tor_assert(is_socks4a);
+  tor_assert(drain_out);
+
```

Comment:
We should initialize drain_out to 0, so that it can't wind up un-set.

Reply:
Fixed in bfa4a7da261b4acb75878522d70c305378c1753e.